src/libmongoc/src/mongoc/mongoc-server-monitor.c

```diff
@@ -319,11 +319,56 @@ _server_monitor_polling_hello (mongoc_server_monitor_t *server_monitor,
    bson_copy_to (hello, &cmd);
 
    _server_monitor_append_cluster_time (server_monitor, &cmd);
+
    ret = _server_monitor_send_and_recv_opquery (
       server_monitor, &cmd, hello_response, error);
    bson_destroy (&cmd);
    return ret;
 }
+static bool
+_server_monitor_awaitable_hello_send_msg (mongoc_server_monitor_t *server_monitor,
+                                      bson_t *cmd,
+                                      bson_error_t *error)
+{
+   mongoc_rpc_t rpc = {0};
+   mongoc_array_t array_to_write;
+   mongoc_iovec_t *iovec;
+   int niovec;
+
+   rpc.header.msg_len = 0;
+   rpc.header.request_id = server_monitor->request_id++;
+   rpc.header.response_to = 0;
+   rpc.header.opcode = MONGOC_OPCODE_MSG;
+   rpc.msg.flags = 0; 
+   rpc.msg.n_sections = 1;
+   rpc.msg.sections[0].payload_type = 0;
+   rpc.msg.sections[0].payload.bson_document = bson_get_data (cmd);
+
+   _mongoc_array_init (&array_to_write, sizeof (mongoc_iovec_t));
+   _mongoc_rpc_gather (&rpc, &array_to_write);
+
+   iovec = (mongoc_iovec_t *) array_to_write.data;
+   niovec = array_to_write.len;
+   _mongoc_rpc_swab_to_le (&rpc);
+
+   MONITOR_LOG (server_monitor,
+                "sending with timeout %" PRId64,
+                server_monitor->connect_timeout_ms);
+
+   if (!_mongoc_stream_writev_full (server_monitor->stream,
+                                    iovec,
+                                    niovec,
+                                    server_monitor->connect_timeout_ms,
+                                    error)) {
+      MONITOR_LOG_ERROR (
+         server_monitor, "failed to write awaitable hello: %s", error->message);
+      _mongoc_array_destroy (&array_to_write);
+      return false;
+   }
```

Comment:
This also needs to receive the reply. Compare with `_server_monitor_send_and_recv_opquery`.

Reply:
+1 done-- good catch!